R/aaa_models.R

```diff
@@ -298,128 +330,131 @@ check_mode_with_no_engine <- function(cls, mode, call) {
 
 check_arg_val <- function(arg) {
   if (rlang::is_missing(arg) || length(arg) != 1 || !is.character(arg))
-    rlang::abort("Please supply a character string for the argument.")
+    cli::cli_abort(
+      "Please supply a character string for the argument. \\
+      not {.obj_type_friendly {arg}}"
+    )
   invisible(NULL)
 }
 
-check_submodels_val <- function(has_submodel) {
+check_submodels_val <- function(has_submodel, call) {
   if (!is.logical(has_submodel) || length(has_submodel) != 1) {
-    rlang::abort("The `submodels` argument should be a single logical.")
+    cli::cli_abort(
+      "The {.arg submodels} argument should be a single logical. \\
+      not {.obj_type_friendly {has_submodel}}"
+    )
   }
   invisible(NULL)
 }
 
 check_func_val <- function(func) {
-  msg <-
-    paste(
-      "`func` should be a named vector with element 'fun' and the optional ",
-      "elements 'pkg', 'range', 'trans', and 'values'.",
-      "`func` and 'pkg' should both be single character strings."
-    )
+  msg <- "{.arg func} should be a named vector with element {.field fun} and \\
+          the optional elements {.field pkg}, {.field range}, {.field trans}, \\
+          and {.field values}. {.field func} and {.field pkg} should both be \\
+          single character strings."
 
   if (rlang::is_missing(func) || !is.vector(func))
-    rlang::abort(msg)
+    cli::cli_abort(msg)
 
   nms <- sort(names(func))
 
   if (all(is.null(nms)))  {
-    rlang::abort(msg)
+    cli::cli_abort(msg)
   }
 
   if (length(func) == 1) {
     if (isTRUE(any(nms != "fun"))) {
-      rlang::abort(msg)
+      cli::cli_abort(msg)
     }
   } else {
     # check for extra names:
     allow_nms <- c("fun", "pkg", "range", "trans", "values")
     nm_check <- nms %in% c("fun", "pkg", "range", "trans", "values")
     not_allowed <- nms[!(nms %in% allow_nms)]
     if (length(not_allowed) > 0) {
-      rlang::abort(msg)
+      cli::cli_abort(msg)
     }
   }
 
   if (!is.character(func[["fun"]])) {
-    rlang::abort(msg)
+    cli::cli_abort(msg)
   }
   if (any(nms == "pkg") && !is.character(func[["pkg"]])) {
-    rlang::abort(msg)
+    cli::cli_abort(msg)
   }
 
   invisible(NULL)
 }
 
 check_fit_info <- function(fit_obj) {
   if (is.null(fit_obj)) {
-    rlang::abort("The `fit` module cannot be NULL.")
+    cli::cli_abort("The {.arg value} argument cannot be NULL.")
```

Comment:
```suggestion
    cli::cli_abort("The {.arg fit_obj} argument cannot be NULL.")
```

Or is this referencing a `caller_arg()` from a frame above?